src/dotty/tools/dotc/transform/TypeSpecializer.scala

```diff
@@ -0,0 +1,284 @@
+package dotty.tools.dotc.transform
+
+import dotty.tools.dotc.ast.{tpd, TreeTypeMap}
+import dotty.tools.dotc.ast.Trees._
+import dotty.tools.dotc.core.Contexts.Context
+import dotty.tools.dotc.core.Decorators.StringDecorator
+import dotty.tools.dotc.core.DenotTransformers.InfoTransformer
+import dotty.tools.dotc.core.Names.Name
+import dotty.tools.dotc.core.Symbols.Symbol
+import dotty.tools.dotc.core.{Symbols, Flags}
+import dotty.tools.dotc.core.Types._
+import dotty.tools.dotc.transform.TreeTransforms.{TransformerInfo, MiniPhaseTransform}
+import scala.collection.mutable
+import dotty.tools.dotc.core.StdNames.nme
+import dotty.tools._
+
+class TypeSpecializer extends MiniPhaseTransform  with InfoTransformer {
+
+  import tpd._
+  override def phaseName = "specialize"
+
+  final var maxTparamsToSpecialize = 0
+
+  private final def specialisedTypeToSuffix(implicit ctx: Context) =
+    Map(defn.ByteType -> "B",
+      defn.BooleanType -> "Z",
+      defn.ShortType -> "S",
+      defn.IntType -> "I",
+      defn.LongType -> "J",
+      defn.FloatType -> "F",
+      defn.DoubleType -> "D",
+      defn.CharType -> "C",
+      defn.UnitType -> "V",
+      defn.AnyRefType -> "L")
```

Comment:
Thanks for pointing this out! I'll take care of changing this when I reach Geneva later today.